modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/reconnection/ReconnectionTestCase.java

```diff
@@ -88,6 +89,23 @@ public void reconnectSource() throws Exception {
     });
   }
 
+  @Test
+  public void doNotStartSourceTwiceAfterExceptionOnReconnection() throws Exception {
+    ((Startable) getFlowConstruct("otherReconnectForever")).start();
+    check(5000, 1000, () -> !capturedEvents.isEmpty());
+    switchOtherFail();
+
+    checkNot(30000, 1000, () -> {
+      synchronized (capturedEvents) {
+        return capturedEvents.stream()
+            .map(event -> (Integer) event.getMessage().getPayload().getValue())
+            .filter(startedSources -> startedSources > 1)
```

Comment:
Is there any way to also check that reconnection was triggered twice and simultaneously? From what I understand this test can also pass if only 1 reconnection is made (which is unlikely due to the Thread sleeps but could also happen if code is changed later on in the Extension Message Source).

Reply:
That is what this test does. The variable `startedSources` is counting how many sources are starting at the same time. I will rename it to something like `currentStartedSources`

Reply:
Done.